### PR TITLE
LA-142 Account Details

### PR DESCRIPTION
**Pull request by imGok**
image

TODO : Url

### CODE REVIEW

File: domain/lib/src/model/user/user.dart [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  final UserId userId;
+  final String locale;
+  final String externalMailLocale;
+  final String domain;
+  final String firstName;
+  final String lastName;
+  final String mail;
+  final bool canUpload;
+  final bool canCreateGuest;
+  final String accountType;
```
Comment (manhduc-nguyen) [inline domain/lib/src/model/user/user.dart:45]: We already have `AccountType` enum, can we reuse in this case ?

File: domain/lib/src/model/user/user.dart [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  final String quotaUuid;
```
Comment (manhduc-nguyen) [inline domain/lib/src/model/user/user.dart:46]: how about create a `QuotaUuid` class

File: domain/lib/src/model/user/user.dart [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+
+import 'package:domain/domain.dart';
+import 'package:equatable/equatable.dart';
+
+class User extends Equatable {
```
Comment (manhduc-nguyen) [inline domain/lib/src/model/user/user.dart:35]: no different in this case, but we should use `class User with EquatableMixin` for best practice, easy to extend `User` class later if we need to.

File: data/lib/src/network/model/response/user_response.dart [hunk 4]
```diff
@@ -31,13 +31,14 @@
...
+import 'package:equatable/equatable.dart';
 import 'package:json_annotation/json_annotation.dart';
 
-part 'user.g.dart';
+part 'user_response.g.dart';
 
 @JsonSerializable()
-class User {
-  User({
+class UserResponse extends Equatable {
```
Comment (manhduc-nguyen) [inline data/lib/src/network/model/response/user_response.dart:40]: same, we should use `with EquatableMixin`

File: data/lib/src/datasource_impl/document_datasource_impl.dart [hunk 5]
```diff
@@ -13,7 +13,7 @@
 // the words “You are using the Free and Open Source version of LinShare™, powered by
 // Linagora © 2009–2020. Contribute to Linshare R&D by subscribing to an Enterprise
 // offer!”. You must also retain the latter notice in all asynchronous messages such as
-// e-mails sent with the Program, (ii) retain all hypertext links between LinShare and
```
Comment (hoangdat) [inline data/lib/src/datasource_impl/document_datasource_impl.dart:16]: any mistake for this?

File: lib/presentation/widget/account_details/account_details_widget.dart [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+              ],
+            )
+          ),
+        ]
+      )
+    );
+  }
+  
+
+  Widget buildAccountDetailsTile(String categoryName, String value) {
```
Comment (manhduc-nguyen) [inline lib/presentation/widget/account_details/account_details_widget.dart:120]: please private this method

File: lib/presentation/widget/initialize_get_it/initialize_viewmodel.dart [hunk 7]
```diff
@@ -73,20 +76,36 @@ class InitializeViewModel extends BaseViewModel {
...
 
   ThunkAction<AppState> getCredentialSuccessAction(CredentialViewState success) {
     return (Store<AppState> store) async {
       _dynamicUrlInterceptors.changeBaseUrl(success.baseUrl.origin);
       _retryInterceptors.setPermanentToken(success.token);
+      store.dispatch(getAuthorizedUserAction());
+    };
+  }
+
+  ThunkAction<AppState> getAuthorizedUserAction() {
```
Comment (manhduc-nguyen) [inline lib/presentation/widget/initialize_get_it/initialize_viewmodel.dart:92]: pls private these methods too.

File: lib/presentation/widget/initialize_get_it/initialize_viewmodel.dart [hunk 8]
```diff
@@ -73,20 +76,36 @@ class InitializeViewModel extends BaseViewModel {
...
+    return (Store<AppState> store) async {
+      await _getAuthorizedInteractor.execute().then((result) => result.fold(
+        (failure) => store.dispatch(authenticationFailureAction()),
```
Comment (hoangdat) [inline lib/presentation/widget/initialize_get_it/initialize_viewmodel.dart:95]: what happen when we can not get authorized user? navigate to what? ...

File: domain/lib/src/model/user/user.dart (hunk 1 above)
Comment (hoangdat) [reply domain/lib/src/model/user/user.dart:45]: yes, please use that,

File: domain/lib/src/model/user/user.dart (hunk 2 above)
Comment (hoangdat) [reply domain/lib/src/model/user/user.dart:46]: good point

File: lib/presentation/widget/initialize_get_it/initialize_viewmodel.dart (hunk 8 above)
Comment (imGok) [reply lib/presentation/widget/initialize_get_it/initialize_viewmodel.dart:95]: ``authenticationFailureAction()`` will navigate to login

File: domain/lib/src/model/user/user.dart (hunk 1 above)
Comment (imGok) [reply domain/lib/src/model/user/user.dart]: Fixed check again pls

File: domain/lib/src/model/user/user.dart (hunk 2 above)
Comment (imGok) [reply domain/lib/src/model/user/user.dart]: Fixed check again pls

File: domain/lib/src/model/user/user.dart (hunk 3 above)
Comment (imGok) [reply domain/lib/src/model/user/user.dart]: Fixed check again pls

File: data/lib/src/network/model/response/user_response.dart (hunk 4 above)
Comment (imGok) [reply data/lib/src/network/model/response/user_response.dart]: Fixed check again pls

File: lib/presentation/widget/account_details/account_details_widget.dart (hunk 6 above)
Comment (imGok) [reply lib/presentation/widget/account_details/account_details_widget.dart]: Fixed check again pls

File: lib/presentation/widget/initialize_get_it/initialize_viewmodel.dart (hunk 7 above)
Comment (imGok) [reply lib/presentation/widget/initialize_get_it/initialize_viewmodel.dart]: Yes sure

File: data/lib/src/datasource_impl/document_datasource_impl.dart (hunk 5 above)
Comment (imGok) [reply data/lib/src/datasource_impl/document_datasource_impl.dart]: Oups sorry

**Review by hoangdat — COMMENTED** [review]
This is the lesson for all of us, when we want to add more function to the product without the care to the priority. This ticket will be stayed at least one more day, and all other issues related to quota may not be resolved on time. @manhduc-nguyen @imGok.
I propose the next step for this ticket:
- keep it simple as much as possible
- I only care to the basic information except (Last Login - Quota)
- Please hide the url also.

File: domain/lib/src/usecases/account/get_account_details_interactor.dart [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+import 'package:domain/src/usecases/account/account_view_state.dart';
+
+class GetAccountDetailsInteractor {
+  final GetAuthorizedInteractor _getAuthorizedInteractor;
+  final GetQuotaInteractor _getQuotaInteractor;
+
+  GetAccountDetailsInteractor(this._getAuthorizedInteractor, this._getQuotaInteractor);
+
+  Future<Either<Failure, Success>> execute() async {
```
[the remark this answers is not in the document]
Comment (manhduc-nguyen) [reply domain/lib/src/usecases/account/get_account_details_interactor.dart:44]: agree, we can streaming user info first, then quota info later

File: lib/presentation/widget/initialize_get_it/initialize_viewmodel.dart [hunk 10]
```diff
@@ -70,23 +73,39 @@ class InitializeViewModel extends BaseViewModel {
...
+        (right) => store.dispatch(_getCredentialSuccessAction((right)))));
     };
   }
 
-  ThunkAction<AppState> getCredentialSuccessAction(CredentialViewState success) {
+  ThunkAction<AppState> _getCredentialSuccessAction(CredentialViewState success) {
     return (Store<AppState> store) async {
       _dynamicUrlInterceptors.changeBaseUrl(success.baseUrl.origin);
       _retryInterceptors.setPermanentToken(success.token);
+      store.dispatch(_getAuthorizedUserAction());
```
Comment (manhduc-nguyen) [inline lib/presentation/widget/initialize_get_it/initialize_viewmodel.dart:88]: `InitializeViewModel ` is just a bridge to navigate between `LoginWidget` and `HomeWidget`, so I really want to keep it simple, less logic as possible, So I think you should get authorized user in `HomeViewModel`. WDYT?

**Comment by imGok** [comment]
Have removed Quota and Account Details thing, review again pls

**Comment by manhduc-nguyen** [comment]
@imGok LGTM

**Review by hoangdat — APPROVED** [review]

**Review by manhduc-nguyen — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**